File: src/main/java/us/illyohs/civilmagiks/common/tile/node/TileManaSpring.java
package us.illyohs.civilmagiks.common.tile.node;

import net.minecraft.nbt.NBTTagCompound;

import net.minecraft.util.ITickable;
import us.illyohs.civilmagiks.api.mana.ManaType;
import us.illyohs.civilmagiks.common.core.util.ManaUtils;
import us.illyohs.civilmagiks.common.core.util.tile.BaseTile;

public class TileManaSpring extends BaseTile implements ITickable {

    public ManaType manaType;

    @Override
    public void update() {
        if (!worldObj.isRemote) {
//            System.out.println(getManaType().getLocalizedName());
        }
    }

    @Override
    public void readFromModNBT(NBTTagCompound mNBT) {

    }

    @Override
    public void writeToModNBT(NBTTagCompound mNBT) {
        mNBT.setString("manatype", getManaType().getLocalizedName());
    }

    public ManaType getManaType() {
        return ManaUtils.getManaFromPos(getWorld(), getPos());
    }

}

File: src/main/scala/us/illyohs/civilmagiks/common/tile/TileManaSpring.scala
package us.illyohs.civilmagiks.common.tile

import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.ITickable
import us.illyohs.civilmagiks.common.core.util.ManaTile
import us.illyohs.civilmagiks.common.core.util.mana.ManaUtils

class TileManaSpring extends ManaTile with ITickable {


//  setManaType(ManaUtils.getManaFromPos(worldObj, pos))

  override def update(): Unit = {

  }

  override def readFromModNBT(mNBT: NBTTagCompound) {

  }

  override def writeToModNBT(mNBT: NBTTagCompound): NBTTagCompound = {
    mNBT.setString("ManaType", getManaType.getLocalizedName)

    mNBT
  }

}
